### PR TITLE
MH-12576 start task failing

### PR DESCRIPTION
**Pull request by KatrinIhler**
This patch changes the behavior of the submit button in the Actions -> Start Task dialog: Instead of waiting for the response of the TasksEndpoint before closing the dialog (which could freeze the ui), this is handed over to a new window and the dialog is closed. The new window can be closed at any time if the user does not want to wait for the backend response. If he does want to wait, then upon response the window displays for each event if the workflow could be started successfully and the error message if it couldn't.

As this patch provides a partial workaround for the ui-freezing-issue, it conflicts with pullrequest #27 which provides another workaround. For easier comparison and discussion this pullrequest was created prematurely, it should not be merged since affected test are not updated yet.

EDIT: This PR builds upon the changes introduced in Christian's PR at Bitbucket which can be found here: https://bitbucket.org/opencast-community/opencast/pull-requests/1845/mh-12576-start-tasks-will-fail-if-start-of/diff

### CODE REVIEW

File: modules/admin-ui/src/main/java/org/opencastproject/adminui/endpoint/TasksEndpoint.java [hunk 1]
```diff
@@ -98,6 +105,9 @@
 
   private Workspace workspace;
 
+  /** A parser for handling JSON documents inside the body of a request. **/
+  private final JSONParser parser = new JSONParser();
```
Comment (staubesv) [inline modules/admin-ui/src/main/java/org/opencastproject/adminui/endpoint/TasksEndpoint.java:109]: Is JSONParser thread-safe?

**Comment by staubesv** [comment]
Info: I accidentally pressed the wrong button... I will not review this as it was developed on behalf of SWITCH (but I can't see a way to unassign myself from being a reviewer...)

**Comment by staubesv** [comment]
@KatrinIhler As this PR changes the UI: Could you please add some screenshots displaying the changes? We once agreed on this being a best practise...

File: modules/admin-ui/src/main/java/org/opencastproject/adminui/endpoint/TasksEndpoint.java [hunk 2]
```diff
@@ -166,23 +180,29 @@ public Response createNewTask(@FormParam("metadata") String metadata) throws Not
...
-          confKeyIter.remove();
-        }
+      return RestUtil.R.badRequest("No eventIds set");
+
+    JSONObject configuration = (JSONObject) metadataJson.get("configuration");
+    Stream<Entry<String, String>> options = Stream.empty();
+    if (configuration != null) {
+      try {
+        options = options.append(JSONUtils.toMap(
+                new org.codehaus.jettison.json.JSONObject(configuration.toJSONString())).entrySet());
```
Comment (karendolan) [inline modules/admin-ui/src/main/java/org/opencastproject/adminui/endpoint/TasksEndpoint.java:192]: Still has that org.codehaus.jettison.json here.

**Comment by KatrinIhler** [comment]
@staubesv While creating the screenshots I found a bug (probably caused by me rebasing on develop), so I'm gonna take care of that tomorrow and then supply screenshots.

@staubesv @karendolan As for the json handling: Apparently Christian had switched over to gson for threadsafety but didn't include it in the pr I took over, but he committed it now and I'm gonna try to merge it into this pr tomorrow. That should hopefully take care of that. Thanks for pointing that out!

**Comment by KatrinIhler** [comment]
@staubesv Screenshots as requested:

before-submit
after-submit
started-successfully
some-failed-to-start

**Comment by gregorydlogan** [comment]
@staubesv noticed your comment on #27, should this go on hold as well since it's solving a similar problem?  Or should we decline #27 in favour of this one?

**Comment by gregorydlogan** [comment]
Talked to @staubesv, this is on hold for the same reason that #27 is on hold!

**Comment by lkiesow** [comment]
From the recent discussion with @staubesv, I understand that pull requests #40 and pull request #27 (both from SWITCH; both basically blocking each other) shall be reviewed, possibly combined and re-submitted in the near future Because of that, is there any reason to not decline this for now?

**Comment by gregorydlogan** [comment]
I was not aware that that was the plan.  I'm fine with declining for now for when it's ready to go.

**Comment by staubesv** [comment]
@KatrinIhler I'm closing this for now. We can do a new PR once we continue to work on this.

**Pull request closed without merging**